stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -815,24 +828,53 @@ extension _JSONEncoder {
         }
     }
 
-    fileprivate func box<T : Encodable>(_ value: T) throws -> NSObject {
+    fileprivate func box(_ value: Encodable) throws -> NSObject {
         return try self.box_(value) ?? NSDictionary()
     }
 
+    fileprivate func box(_ dict: [String : Encodable]) throws -> NSObject? {
+        let depth = self.storage.count
+        let result = self.storage.pushKeyedContainer()
+        do {
+            for (k, v) in dict {
```

Comment:
Slight style nit: we prefer `(key, value)` to `(k, v)` where possible (when names don't conflict)